### ISSUE TITLE
Question regarding frame construction

### ISSUE THREAD

**Issue by goerch** [JuliaDebug/JuliaInterpreter.jl#502]
When working on [this issue](https://github.com/aviatesk/JET.jl/pull/278) we noticed differences between `JuliaInterpreter` optimized code and `JET` code. We therefore changed `Frame(mod, src)` to 

``` 
    # NOTE if `JuliaInterpreter.optimize!` may modify `src`, `src` and `concretize` can be inconsistent
    # here we create `JuliaInterpreter.Frame` by ourselves disabling the optimization (#277)
    framecode = JuliaInterpreter.FrameCode(mod, src, optimize=false)
    @assert length(framecode.src.code) == length(concretize)
    frame = Frame(framecode, JuliaInterpreter.prepare_framedata(framecode, Any[]))
```

using internal `JuliaInterpreter` code. To be on the safe side I'd like to propose a PR to extend the `Frame` constructor with the parameters for `generator` and `optimize`, if there is no existing way?

**Comment by KristofferC** [comment]
Seems reasonable to me.

**Comment by timholy** [comment]
Closed by #503
